internal/sql/service.go

```diff
@@ -55,11 +55,12 @@ func (s Service) ExecuteStatement(ctx context.Context, stmt sql.Statement) (sql.
 	if ctx, err = updateContextWithOptions(ctx, stmt); err != nil {
 		return &Result{}, nil
 	}
-	params := make([]driver.Value, len(stmt.Parameters))
-	for i := range stmt.Parameters {
-		params[i] = stmt.Parameters[i]
+	stmtParams := stmt.Parameters()
+	params := make([]driver.Value, len(stmtParams))
+	for i := range stmtParams {
+		params[i] = stmtParams[i]
```

Comment:
Since `driver.Value` is an `interface{}`, you can use the following trick to convert `[]interface{}` to `[]driver.Value` without creating another slice:
```
params := *(*([]driver.Value))(unsafe.Pointer(&stmtParams))
```

Reply:
Thanks for sharing that. It is replaced https://github.com/hazelcast/hazelcast-go-client/pull/755/commits/ac53752ddf6f6576f5f2955ebb21b2aaff1795f0